app/src/main/java/com/concordia/mcga/lib/BuildingBottomSheetInfo.java

```diff
@@ -0,0 +1,1168 @@
+         * Called when the user's input indicates that they want to capture
+         * the given child view with the pointer indicated by pointerId. The
+         * callback should return true if the user is permitted to drag the
+         * given view with the indicated pointer.
+         * @param child
+         * @param pointerId
+         * @return true if the user is permitted to drag the given view with the indicated pointer.
+         */
+        @Override
+        public boolean tryCaptureView(View child, int pointerId ) {
+            if ( mState == STATE_DRAGGING ) {
+                return false;
+            }
+            if ( mTouchingScrollingChild ) {
+                return false;
+            }
+            if ( mState == STATE_EXPANDED  &&  mActivePointerId == pointerId ) {
+                View scroll = mNestedScrollingChildRef.get();
+                if (scroll != null && ViewCompat.canScrollVertically(scroll, -1)) {
+                    // Let the content scroll up
+                    return false;
+                }
+            }
+            return mViewRef != null && mViewRef.get() == child;
+        }
+
+        /**
+         * Called when the captured view's position changes as the result of a drag or settle.
+         * @param changedView
+         * @param left
+         * @param top
+         * @param dx
+         * @param dy
+         */
+        @Override
+        public void onViewPositionChanged(View changedView, int left, int top, int dx, int dy ) {
+            dispatchOnSlide( top );
+        }
+
+        /**
+         * Called when the drag state changes. See the STATE_* constants for more information.
+         * @param state
+         */
+        @Override
+        public void onViewDragStateChanged( int state ) {
+            if ( state == ViewDragHelper.STATE_DRAGGING ) {
+                setStateInternal( STATE_DRAGGING );
+            }
+        }
+
+        /**
+         * Called when the child view is no longer being actively dragged.
+         * The fling velocity is also supplied, if relevant. The velocity
+         * values may be clamped to system minimums or maximums.
+         * @param releasedChild
+         * @param xvel
+         * @param yvel
+         */
+        @Override
+        public void onViewReleased(View releasedChild, float xvel, float yvel ) {
+            int top;
+            @State int targetState;
+            if ( yvel < 0 ) { // Moving up
+                top = mMinOffset;
+                targetState = STATE_EXPANDED;
+            }
+            else
+            if ( mHideable  &&  shouldHide(releasedChild, yvel) ) {
+                if (mType == "building_information"){
+                    top = mParentHeight;
+                    targetState = STATE_HIDDEN;
+                }else {
+                    top = mMaxOffset;
+                    targetState = STATE_COLLAPSED;
+                }
+            }
+            else
+            if ( yvel == 0.f ) {
```

Comment:
🚨**This is a critical bug**. 🚨 Equality tests should **not** be made with floating point values.
Use an epsilon, (NUMERICAL METHODS)

`float epsilon = 1e-2; //or some other  minimum value  relevant to our application`
`if ( Math.abs(yvel) < epsilon){...}`
 
This introduces erratic behaviour for the bottomSheet because we can never check for the velocity to be *exactly* `0.f`, it can be close to zero but exactly zero is almost impossible due to floating point math.